src/Groups/sub.jl

```diff
@@ -52,18 +55,22 @@ julia> H == alternating_group(4)
 true
 ```
 """
-function sub(G::GAPGroup, gens::AbstractVector{S}) where S <: GAPGroupElem
+function sub(G::GAPGroup, gens::AbstractVector{S}; check::Bool = true) where S <: GAPGroupElem
   @assert elem_type(G) == S
   elems_in_GAP = GapObj([x.X for x in gens])
-  H = GAP.Globals.Subgroup(G.X,elems_in_GAP)::GapObj
+  if check
+    H = GAP.Globals.Subgroup(G.X,elems_in_GAP)::GapObj
+  else
+    H = GAP.Globals.SubgroupNC(G.X,elems_in_GAP)::GapObj
+  end
   return _as_subgroup(G, H)
 end
 
 function sub(gens::GAPGroupElem...)
    length(gens) > 0 || throw(ArgumentError("Empty list"))
    l = collect(gens)
    @assert all(x -> parent(x) == parent(l[1]), l)
```

Comment:
Can you do such a check also in the `sub(G, gens)` case above and disable the check automatically?

Reply:
@thofma I do not understand what you want to suggest.
Shall the `sub(G, gens)` method check that all elements in `gens` have the same parent, or that their common parent is `G`? If yes then this would be more restrictive than what is used now, several of the current tests would fail.
And under which conditions shall the parent check (here or in the `sub(G, gens)` method?) automatically disabled?

Reply:
If all elements of `gens` have parent equal to `G`, then I would have guessed that one can directly call the `NC` version?

So one can replace the `if check` by
```
if check && any(x -> parent(x) !== G, gens)
```

Which test would start to fail in this case?

Reply:
Now I understand your idea.
Yes, we can argue that an explicit membership test is needed only for those element in `gens` that have parent different from `G`.